application/client.plugins/projects/serial/src/lib/views/dialog/port.available/components.ts

```diff
@@ -0,0 +1,197 @@
+    }
+
+    ngAfterViewInit() {
+        if (!this._ng_isConnected(this.port)) {
+            this._read = 0;
+            this._canvas = document.getElementById(`canvas_${this.port.path}`) as HTMLCanvasElement;
+            this._createChart().then(() => {
+                this._update();
+            }).catch((error: Error) => {
+                this._logger.error(error.message);
+            });
+        }
+
+        this._subscriptions.Subscription = this.observer.event.subscribe((message: any) => {
+            if (typeof message !== 'object' && message === null) {
+                return;
+            }
+            if (message) {
+                this._update();
+            }
+        });
+
+        this._subscriptions.Subscription = Service.getObservable().event.subscribe((message: any) => {
+            if (message[this.port.path]) {
+                    this._read = message[this.port.path];
+            }
+        });
+    }
+
+    ngOnDestroy() {
+        this._spark = [];
+        if (this._chart) {
+            this._chart.destroy();
+        }
+        Object.keys(this._subscriptions).forEach((key: string) => {
+            this._subscriptions[key].unsubscribe();
+        });
+        this._destroyed = true;
+    }
+
+    private _forceUpdate() {
+        if (this._destroyed) {
+            return;
+        }
+        this._cdRef.detectChanges();
+    }
+
+    private _formatLoad(load: number): string {
+        let read: string = '';
+        if (load > 1024 * 1024 * 1024) {
+            read = (load / 1024 / 1024 / 1024).toFixed(2) + ' Gb';
+        } else if (load > 1024 * 1024) {
+            read = (load / 1024 / 1024).toFixed(2) + ' Mb';
+        } else if (load > 1024) {
+            read = (load / 1024).toFixed(2) + ' Kb';
+        } else {
+            read = load + ' b';
+        }
+        return read;
+    }
+
+    public _ng_isConnected(port: IPortInfo): boolean {
+        if (this.connected && this.connected.find(connected => connected.port.path === port.path)) {
+            return true;
+        }
+        return false;
+    }
+
+    public _ng_read(port: IPortInfo) {
+        return this._formatLoad(this.spyState[port.path]);
+    }
+
+    private _color(): number {
+        return Math.round(Math.random() * 255);
+    }
+
+    private _colorize(): string {
+        const rgb: Irgb = {
+            red: this._color(),
+            green: this._color(),
+            blue: this._color(),
+            opacity: 1,
+        };
+        return `rgba(${rgb.red}, ${rgb.green}, ${rgb.blue}, ${rgb.opacity})`;
+    }
+
+    private _createChart(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            this._ctx = this._canvas.getContext('2d');
+            if (this._chart) {
+                reject(`Chart for ${this.port.path} already exists!`);
+            } else {
+                this._chart = new Chart(this._ctx, {
+                    type: 'line',
+                    data: {
```

Comment:
This method doesn't have any asynchronous operations, it might me not a promise, but return Error (in case of error) or undefined (in case of success).